### PR TITLE
Census-Yes-No-Routing

### PR DESCRIPTION
**Pull request by yoganandkunche**
### What is the context of this PR?
This PR is for Census e2e Yes/No routing scenarios.

### How to review 

Run the below test: 

yarn run test_functional -- --spec=census/routing/census-routing-scenarios

Few issues are identified which will be raised as separate issues. Which are currently commented with in this .js file so that the Master wont fail.

### CODE REVIEW

File: tests/functional/pages/surveys/census/household/else-permanent-or-family-home.page.js [hunk 1]
```diff
@@ -12,6 +12,10 @@ class ElsePermanentOrFamilyHomePage extends MultipleChoiceWithOtherPage {
     return this
   }
 
+  getAlertText(year) {
```
Comment (jonnyshaw89) [inline tests/functional/pages/surveys/census/household/else-permanent-or-family-home.page.js]: This should be defined in the base question.page.js

File: tests/functional/pages/surveys/census/household/permanent-or-family-home.page.js [hunk 2]
```diff
@@ -12,6 +12,10 @@ class PermanentOrFamilyHomePage extends MultipleChoiceWithOtherPage {
     return this
   }
 
+  getAlertText(year) {
```
Comment (jonnyshaw89) [inline tests/functional/pages/surveys/census/household/permanent-or-family-home.page.js]: Same as above

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 3]
```diff
@@ -0,0 +1,728 @@
...
+    })
+
+    it('Given I am answering question 3 in the who lives here section, When I select -yes- as the respone, Then I am routed to Who lives here question 4 ', function () {
+        openAndStartCensusQuestionnaire('census_household.json', true)
+        PermanentOrFamilyHome.clickPermanentOrFamilyHomeAnswerYes().submit()
+        HouseholdComposition.setPersonName(0, 'John Smith').addPerson().setPersonName(1, 'Jane Smith').submit()
+        EveryoneAtAddressConfirmation.clickEveryoneAtAddressConfirmationAnswerYes().submit()
+        OvernightVisitors.setOvernightVisitorsAnswer(0).submit()
+    })
+    // issue present error 500
```
Comment (jonnyshaw89) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:127]: Are the issues then duplicating these tests?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js (hunk 3 above)
Comment (yoganandkunche) [reply tests/functional/spec/census/routing/census-routing-scenarios.spec.js:127]: Yes the ones I have commented.

File: tests/functional/pages/surveys/census/household/permanent-or-family-home.page.js (hunk 2 above)
Comment (yoganandkunche) [reply tests/functional/pages/surveys/census/household/permanent-or-family-home.page.js]: Made the changes

File: tests/functional/pages/surveys/census/household/else-permanent-or-family-home.page.js (hunk 1 above)
Comment (yoganandkunche) [reply tests/functional/pages/surveys/census/household/else-permanent-or-family-home.page.js]: Made the changes

**Comment by yoganandkunche** [comment]
@jonnyshaw89 Made the changes.. ready for you to review

**Review by jonnyshaw89 — APPROVED** [review]

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 4]
```diff
@@ -0,0 +1,728 @@
...
+        HouseholdComposition.setPersonName(0, 'John Smith').addPerson().setPersonName(1, 'Jane Smith').submit()
+    })
+
+    it('Given I am answering question 1 in the who lives here section, When I select -no- as the respone, Then I am routed to Who lives here question 2 ', function () {
+        openAndStartCensusQuestionnaire('census_household.json', true)
+        PermanentOrFamilyHome.clickPermanentOrFamilyHomeAnswerNo().submit()
+        ElsePermanentOrFamilyHome.clickElsePermanentOrFamilyHomeAnswerYes()
+    })
+
+    it('Given I am answering question 1 in the who lives here section, When I dont select any respone, Then I a alert msg saying mandatory field must be displayed ', function () {
```
Comment (benjefferies) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:93]: Typo respone  -> response

Comment (benjefferies) [reply tests/functional/spec/census/routing/census-routing-scenarios.spec.js:93]: `Then I a alert msg saying mandatory field must be displayed` -> `Then I get an alert message saying mandatory field must be displayed`

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 5]
```diff
@@ -0,0 +1,728 @@
...
+    })
+
+    it('Given I am answering question 1a in the who lives here section, When I select -no- as the respone, Then I am routed to How many visitors... question 4 ', function () {
+        openAndStartCensusQuestionnaire('census_household.json', true)
+        PermanentOrFamilyHome.clickPermanentOrFamilyHomeAnswerNo().submit()
+        ElsePermanentOrFamilyHome.clickElsePermanentOrFamilyHomeAnswerNo().submit()
+        OvernightVisitors.setOvernightVisitorsAnswer(0).submit()
+    })
+
+    it('Given I am answering question 1a in the who lives here section, When I select -no- as the respone, Then I am routed to How many visitors... question 4 ', function () {
```
Comment (benjefferies) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:113]: Looks like the Then part isn't correct in the title

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 6]
```diff
@@ -0,0 +1,728 @@
...
+    NumberOfVehicles.setNumberOfVehiclesAnswer(2).submit()
+    HouseholdAndAccommodationCompleted.submit()
+
+  // household-member
+    HouseholdMemberBegin.submit()
+    DetailsCorrect.clickDetailsCorrectAnswerNoINeedToChangeMyName().submit()
+    CorrectName.setCorrectNameAnswer('Yoganand Kumar Kunche').submit()
+  })
+
+  //issue present - error 404
```
Comment (LJBabbage) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:194]: If we are not using it, we need to remove it

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 7]
```diff
@@ -0,0 +1,728 @@
...
+    Sex.clickSexAnswerMale().submit()
+    DateOfBirth.setDateOfBirthAnswerDay(2).setDateOfBirthAnswerMonth(3).setDateOfBirthAnswerYear(1980).submit()
+    Over16.clickOver16AnswerYes().submit()
+    MaritalStatus.clickMaritalStatusAnswerMarried().submit()
+    AnotherAddress.clickAnotherAddressAnswerNo().submit()
+    InEducation.clickInEducationAnswerNo().submit()
+    CountryOfBirth.clickCountryOfBirthEnglandAnswerEngland().submit()
+  })
+
+// Issue 404
```
Comment (LJBabbage) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:342]: Same as above

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 8]
```diff
@@ -0,0 +1,728 @@
...
+    DetailsCorrect.clickDetailsCorrectAnswerYesThisIsMyFullName().submit()
+    Sex.clickSexAnswerMale().submit()
+    DateOfBirth.setDateOfBirthAnswerDay(2).setDateOfBirthAnswerMonth(3).setDateOfBirthAnswerYear(1980).submit()
+    Over16.clickOver16AnswerYes().submit()
+    MaritalStatus.clickMaritalStatusAnswerMarried().submit()
+    AnotherAddress.clickAnotherAddressAnswerOther().submit()
+    AddressType.clickAddressTypeAnswerArmedForcesBaseAddress().submit()
+  })
+
+  it('Given I am answering question 7 in the individual detail section - 7. Are you a schoolchild or student in full-time education?, When I dselect -No- as respone, Then I am routed to 9. What is your country of birth?', function () {
```
Comment (benjefferies) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:311]: dselect -> select

**Review by LJBabbage — CHANGES_REQUESTED** [review]

**Review by LJBabbage — APPROVED** [review]

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 9]
```diff
@@ -0,0 +1,728 @@
...
+    MaritalStatus.clickMaritalStatusAnswerMarried().submit()
+    AnotherAddress.clickAnotherAddressAnswerNo().submit()
+    InEducation.clickInEducationAnswerYes().submit()
+    TermTimeLocation.clickTermTimeLocationAnswerAtThisAddress().submit()
+    CountryOfBirth.clickCountryOfBirthEnglandAnswerEngland().submit()
+    Carer.clickCarerAnswerYes50OrMoreHoursAWeek().submit()
+    NationalIdentity.clickNationalIdentityAnswerScottish().submit()
+  })
+
+  it('Given I am answering question 25. Thinking of the last 12 months, have you..., When I select -No- as respone, Then I am routed to 26. Last week were you:', function () {
```
Comment (benjefferies) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:541]: Does this make sense in what it's doing?

To me it feels like it should be

`Given I am answering question 26. 26. Last week were you:, when I select working as an employees, then I should be on household complete section

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 10]
```diff
@@ -0,0 +1,728 @@
...
+      PastUsualAddress.clickPastUsualAddressAnswerThisAddress().submit()
+      Passports.clickPassportsAnswerUnitedKingdom().submit()
+      Disability.clickDisabilityAnswerNo().submit()
+      Qualifications.clickQualificationsAnswerUndergraduateDegree().submit()
+      Volunteering.clickVolunteeringAnswerNo().submit()
+      EmploymentType.clickEmploymentTypeAnswerWorkingAsAnEmployee().submit()
+      HouseholdMemberCompleted.submit()
+    })
+
+    it('Given I am answering question 25. Thinking of the last 12 months, have you..., When I select -Yes, at least once a week- as respone, Then I am routed to 26. Last week were you:', function () {
```
Comment (benjefferies) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:588]: See comment above

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 11]
```diff
@@ -0,0 +1,728 @@
...
+        PastUsualAddress.clickPastUsualAddressAnswerThisAddress().submit()
+        Passports.clickPassportsAnswerUnitedKingdom().submit()
+        Disability.clickDisabilityAnswerNo().submit()
+        Qualifications.clickQualificationsAnswerUndergraduateDegree().submit()
+        Volunteering.clickVolunteeringAnswerYesAtLeastOnceAWeek().submit()
+        EmploymentType.clickEmploymentTypeAnswerWorkingAsAnEmployee().submit()
+        HouseholdMemberCompleted.submit()
+      })
+
+      it('Given I am answering question 25. Thinking of the last 12 months, have you..., When I select -Yes, less than once a week but at least once a month-, at least once a week- as respone, Then I am routed to 26. Last week were you:', function () {
```
Comment (benjefferies) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:635]: See comment above

File: tests/functional/spec/census/routing/census-routing-scenarios.spec.js [hunk 12]
```diff
@@ -0,0 +1,728 @@
...
+          PastUsualAddress.clickPastUsualAddressAnswerThisAddress().submit()
+          Passports.clickPassportsAnswerUnitedKingdom().submit()
+          Disability.clickDisabilityAnswerNo().submit()
+          Qualifications.clickQualificationsAnswerUndergraduateDegree().submit()
+          Volunteering.clickVolunteeringAnswerYesLessThanOnceAWeekButAtLeastOnceAMonth().submit()
+          EmploymentType.clickEmploymentTypeAnswerWorkingAsAnEmployee().submit()
+          HouseholdMemberCompleted.submit()
+        })
+
+      it('Given I am answering question 25. Thinking of the last 12 months, have you..., When I select -Yes, less often-, at least once a week- as respone, Then I am routed to 26. Last week were you:', function () {
```
Comment (benjefferies) [inline tests/functional/spec/census/routing/census-routing-scenarios.spec.js:682]: See comment above

**Pull request merged**